packages/headless/src/features/pagination/pagination-slice.ts

```diff
@@ -70,6 +82,30 @@ export const paginationReducer = createReducer(
       .addCase(executeSearch.fulfilled, (state, action) => {
         const {response} = action.payload;
         state.totalCountFiltered = response.totalCountFiltered;
+      })
+      .addCase(deselectAllFacetValues, (state) => {
```

Comment:
Did not include `deselectAllDateFacetValues` and `deselectAllNumericFacetValues` because they are the same as  the `deselectAllFacetValues` action.